tests/multipleBackend/routes/routeBackbeat.js

```diff
@@ -274,6 +303,982 @@ describe.skip('backbeat routes', () => {
+                    next),
+                next => s3.putObject({ Bucket: bucket, Key: keyName, Body: new Buffer(testData) }, next),
+                next => makeBackbeatRequest({
+                    method: 'GET',
+                    resourceType: 'metadata',
+                    bucket,
+                    objectKey: keyName,
+                    queryObj: {
+                        versionId: 'null',
+                    },
+                    authCredentials: backbeatAuthCredentials,
+                }, (err, data) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    const { error, result } = updateStorageClass(data, storageClass);
+                    if (error) {
+                        return next(error);
+                    }
+                    objMD = result;
+                    return next();
+                }),
+                next => s3.deleteObject({ Bucket: bucket, Key: keyName, VersionId: 'null' }, next),
+                next => makeBackbeatRequest({
+                    method: 'PUT',
+                    resourceType: 'metadata',
+                    bucket,
+                    objectKey: keyName,
+                    queryObj: {
+                        versionId: 'null',
+                    },
+                    authCredentials: backbeatAuthCredentials,
+                    requestBody: objMD.getSerialized(),
+                }, next),
+                next => s3.headObject({ Bucket: bucket, Key: keyName }, next),
+                next => s3.listObjectVersions({ Bucket: bucket }, next),
+            ], (err, data) => {
+                if (err) {
+                    return done(err);
+                }
+                const headObjectRes = data[6];
+                assert.strictEqual(headObjectRes.VersionId, 'null');
+                assert.strictEqual(headObjectRes.StorageClass, storageClass);
+
+                const listObjectVersionsRes = data[7];
+                const { DeleteMarkers, Versions } = listObjectVersionsRes;
+
+                assert.strictEqual(DeleteMarkers.length, 0);
+                assert.strictEqual(Versions.length, 1);
+
+                const currentVersion = Versions[0];
+                assert(currentVersion.IsLatest);
+
+                assertVersionIsNullAndUpdated(currentVersion);
+
+                return done();
+            });
+        });
+
+        it('should create a new null version if versioning suspended and delete marker null version', done => {
+            let objMD;
+            return async.series([
+                next => s3.putBucketVersioning({ Bucket: bucket, VersioningConfiguration: { Status: 'Enabled' } },
+                    next),
+                next => s3.putBucketVersioning({ Bucket: bucket, VersioningConfiguration: { Status: 'Suspended' } },
+                    next),
+                next => s3.putObject({ Bucket: bucket, Key: keyName, Body: new Buffer(testData) }, next),
```

Comment:
There's probably no need to PUT an object first before DELETE, as a DELETE would normally put a delete marker with a null version.